File: Quick Starts/Security/CS-UnityIntegration/SecurityQuickStart/QuickstartChildForms.cs
//===============================================================================
// Microsoft patterns & practices Enterprise Library
// Security Application Block QuickStart
//===============================================================================
// Copyright © Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Practices.Unity;

namespace SecurityQuickStart
{
    /// <summary>
    /// This class holds the instances of all the child forms
    /// the main quickstart form requires. This reduces the
    /// size of the main form's constructor parameter list.
    /// </summary>
    public class QuickstartChildForms
    {
        [Dependency]
        public AddUserRoleForm AddUserRoleForm { get; set; }

        [Dependency]
        public UserRoleForm UserRoleForm { get; set; }
        
        [Dependency]
        public UsersForm UsersForm { get; set; }
        
        [Dependency]
        public CredentialsForm CredentialsForm { get; set; }
        
        [Dependency]
        public ProfileForm ProfileForm { get; set; }
     
        [Dependency]
        public RoleAuthorizationForm RoleAuthForm { get; set; }
    }
}

File: Quick Starts/Security/VB-UnityIntegration/SecurityQuickStart/QuickStartChildForms.vb
'===============================================================================
' Microsoft patterns & practices Enterprise Library
' Security Application Block QuickStart
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================

Imports Microsoft.Practices.Unity

Public Class QuickStartChildForms

    Private myAddUserRoleForm As AddUserRoleForm
    Private myCredentialsForm As CredentialsForm
    Private myProfileForm As ProfileForm
    Private myRoleAuthorizationForm As RoleAuthorizationForm
    Private myUserRoleForm As UserRoleForm
    Private myUsersForm As UsersForm

    <Dependency()> _
    Public Property AddUserRoleForm() As AddUserRoleForm
        Get
            Return myAddUserRoleForm
        End Get
        Set(ByVal value As AddUserRoleForm)
            myAddUserRoleForm = value
        End Set
    End Property

    <Dependency()> _
    Public Property CredentialsForm() As CredentialsForm
        Get
            Return myCredentialsForm
        End Get
        Set(ByVal value As CredentialsForm)
            myCredentialsForm = value
        End Set
    End Property

    <Dependency()> _
    Public Property ProfileForm() As ProfileForm
        Get
            Return myProfileForm
        End Get
        Set(ByVal value As ProfileForm)
            myProfileForm = value
        End Set
    End Property

    <Dependency()> _
    Public Property RoleAuthorizationForm() As RoleAuthorizationForm
        Get
            Return myRoleAuthorizationForm
        End Get
        Set(ByVal value As RoleAuthorizationForm)
            myRoleAuthorizationForm = value
        End Set
    End Property

    <Dependency()> _
    Public Property UserRoleForm() As UserRoleForm
        Get
            Return myUserRoleForm
        End Get
        Set(ByVal value As UserRoleForm)
            myUserRoleForm = value
        End Set
    End Property

    <Dependency()> _
    Public Property UsersForm() As UsersForm
        Get
            Return myUsersForm
        End Get
        Set(ByVal value As UsersForm)
            myUsersForm = value
        End Set
    End Property

End Class
